### DISCUSSION TITLE
Anybody has any good resources and/or tips on how it is possible to eliminate Storyboards from UIKit projects?

### DISCUSSION THREAD

**Discussion by pedrohgmuniz** [community/community#36458] [Mobile] [answered]
I haven't been able to find any recent material on the subject :(

**Comment by ameninadogorro** [comment, marked as answer]
Only last week I found this excellent and relatively recent video on the topic: https://www.youtube.com/watch?v=_U6_l58Cv4E Here is a summary of the steps: 1. First simply delete the Main Storyboard file from your projects' files list on the left; 2. Then select the project and go to the Info tab. You have to delete that Main storyboard file base name you'll find on the list; 3. Now just do a search for "storyboard" on the whole project (left side of Xcode) and click on that result; 4. Simply delete that file as well by clicking on the minus sign. Now the last step: 5. Now that we don't have a storyboard anymore, we basically have to tell the app how to launch without it. To do that, we go to our SceneDelegate. According to the video I'm sharing with you, a scene is basically a window in the app. It's something created at the time when iPhones and iPads still didn't have the ability to open more than one app at the same time (displaying them simultaneously, like side by side, for ex.). Ir order to do that, this thing called scene was introduced. In our case, we will tell our scene to launch our app. Go to SceneDelegate and replace the first func for this one:

 ```
func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {

        guard let windowScene = (scene as? UIWindowScene) else { return }

        let navigationController = UINavigationController(rootViewController: NameOfYourFirstScreenViewController())

        window = UIWindow(frame: windowScene.coordinateSpace.bounds)
        window?.windowScene = windowScene

        window?.rootViewController = navigationController
        window?.makeKeyAndVisible()
    }
```
